flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SubSequenceInputTypeStrategy.java

```diff
@@ -0,0 +1,260 @@
+			List<DataType> originalArgumentDataTypes = originalCallContext.getArgumentDataTypes();
+			return originalArgumentDataTypes.subList(
+				split.startIndex,
+				split.endIndex != null ? split.endIndex : originalArgumentDataTypes.size());
+		}
+
+		@Override
+		public Optional<DataType> getOutputDataType() {
+			return originalCallContext.getOutputDataType();
+		}
+	}
+
+	/**
+	 * A Builder for {@link SubSequenceInputTypeStrategy}.
+	 */
+	@PublicEvolving
+	public static class SubSequenceStrategyBuilder {
+		private final List<ArgumentsSplit> argumentsSplits = new ArrayList<>();
+		private int currentPos = 0;
+
+		/**
+		 * Defines that we expect a single argument at the next position.
+		 */
+		public SubSequenceStrategyBuilder argument(ArgumentTypeStrategy argumentTypeStrategy) {
+			SequenceInputTypeStrategy singleArgumentStrategy = new SequenceInputTypeStrategy(
+				Collections.singletonList(argumentTypeStrategy), null);
+			argumentsSplits.add(new ArgumentsSplit(currentPos, currentPos + 1, singleArgumentStrategy));
+			currentPos += 1;
+			return this;
+		}
+
+		/**
+		 * Defines a common {@link InputTypeStrategy} for the next arguments. Given
+		 * input strategy must expect a constant number of arguments. That means that both
+		 * the minimum and maximum number of arguments must be defined and equal to each other.
+		 *
+		 * <p>If you need a varying logic use {@link #finishWithvarying(InputTypeStrategy)}.
+		 */
+		public SubSequenceStrategyBuilder subSequence(InputTypeStrategy inputTypeStrategy) {
+			Optional<Integer> maxCount = inputTypeStrategy.getArgumentCount().getMaxCount();
+			Optional<Integer> minCount = inputTypeStrategy.getArgumentCount().getMinCount();
+			if (!maxCount.isPresent() || !minCount.isPresent() || !maxCount.get().equals(minCount.get())) {
+				throw new IllegalArgumentException("Both the minimum and maximum number of expected arguments must" +
+					" be defined and equal to each other.");
+			}
+			argumentsSplits.add(new ArgumentsSplit(currentPos, currentPos + maxCount.get(), inputTypeStrategy));
+			currentPos += maxCount.get();
+			return this;
+		}
+
+		/**
+		 * Defines a common {@link InputTypeStrategy} for the next arguments. Given
+		 * input strategy must expect a varying number of arguments. That means that the
+		 * maximum number of arguments must not be defined.
+		 */
+		public InputTypeStrategy finishWithvarying(InputTypeStrategy inputTypeStrategy) {
+			ArgumentCount strategyArgumentCount = inputTypeStrategy.getArgumentCount();
+			Optional<Integer> maxCount = strategyArgumentCount.getMaxCount();
+			if (maxCount.isPresent()) {
+				throw new IllegalArgumentException("The maximum number of arguments must not be defined.");
+			}
+			argumentsSplits.add(new ArgumentsSplit(currentPos, null, inputTypeStrategy));
+			int minCount = currentPos;
+			Optional<Integer> strategyMinCount = strategyArgumentCount.getMinCount();
+			if (strategyMinCount.isPresent()) {
```

Comment:
same comment as bevor, why don't we use the fluent `.orElse` instead of `.isPresent()`.